core/src/main/java/hudson/model/listeners/RunListener.java

```diff
@@ -108,6 +109,12 @@ public void onCompleted(R r, @Nonnull TaskListener listener) {}
     public void onFinalized(R r) {}
 
     /**
+     * Called when a Run is entering execution.
+     * @param r
```

Comment:
state is State.BUILDING. fireStarted is actually run later, as the BuildListener has been created. Feel free to suggest a better name.